controllers/controllers.go

```diff
@@ -48,3 +51,49 @@ func (r *instanceReconciler) configureInstance(instance *instances.InstanceInfo,
 	}
 	return nil
 }
+
+// instanceFromNode returns an instance object for the given node. Requires a username that can be used to SSH into the
+// instance to be annotated on the node.
+func (r *instanceReconciler) instanceFromNode(node *core.Node) (*instances.InstanceInfo, error) {
+	if node.Annotations[UsernameAnnotation] == "" {
+		return nil, errors.New("node is missing valid username annotation")
+	}
+	addr, err := getAddress(node.Status.Addresses)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get node address")
+	}
+	return instances.NewInstanceInfo(addr, node.Annotations[UsernameAnnotation], ""), nil
+}
+
+// getAddress returns a non-ipv6 address that can be used to reach a Windows node. This can be either an ipv4
+// or dns address.
+func getAddress(addresses []core.NodeAddress) (string, error) {
+	for _, addr := range addresses {
+		if addr.Type == core.NodeInternalIP || addr.Type == core.NodeInternalDNS {
+			// filter out ipv6
+			if net.ParseIP(addr.Address) != nil && net.ParseIP(addr.Address).To4() == nil {
+				continue
+			}
+			return addr.Address, nil
+		}
+	}
+	return "", errors.New("no usable address")
+}
+
+// removeNode removes the specified node from the cluster
+func (r *instanceReconciler) removeNode(node *core.Node) error {
```

Comment:
should this be deconfigureInstance() like we have configureInstance(), we would have to find the instance info from the node previous to this step and pass it in here in that case.